src/emulator/gui/src/gui.cpp

```diff
@@ -167,18 +167,24 @@ static void init_icons(HostState &host) {
         int32_t width = 0;
         int32_t height = 0;
         vfs::FileBuffer buffer;
+        const std::string default_icon = "../data/image/icon.png";
 
         vfs::read_app_file(buffer, host.pref_path, game.title_id, "sce_sys/icon0.png");
         if (buffer.empty()) {
-            LOG_WARN("Icon not found for title {}, {}.", game.title_id, game.title);
-            continue;
+            if (fs::exists("data/image/icon.png")) {
+                LOG_INFO("Default icon found for title {}, {}.", game.title_id, game.title);
+                vfs::read_file(VitaIoDevice::UX0, buffer, host.base_path, default_icon);
```

Comment:
This looks like contradiction. Please explain:

- you are reading  "../data/image/icon.png", but you check for  "data/image/icon.png" if it exists

Reply:
ah no is same, because using read_file for can use buffer, and need use vitadevice, but both indicate exactly same place

Reply:
this is a dirty hack tbh.

open an ifstream and read into the buffer

Reply:
i have not founded how use ifstream for read buffer :/

Reply:
```cpp
const std::string default_icon = "data/image/icon.png";
std::ifstream image_stream(image_path, std::ios::ate);

const std::size_t fsize = image_stream.tellg();
buffer.resize(fsize);

image_stream.seek(0, std::ios::beg);
image_stream.read(reinterpret_cast<char*>(&buffer[0]), fsize);
```

Reply:
oh thx i look :)

Reply:
Done thx 👍